numpyro/distributions/distribution.py

```diff
@@ -129,24 +130,49 @@ def logpdf(self, x, *args, **kwargs):
 
 
 class jax_discrete(jax_generic, osp_stats.rv_discrete):
-    def _support_mask(self, k):
-        return (k >= self.a) & (k <= self.b) & (np.floor(k) == k)
+    def __new__(cls, *args, **kwargs):
+        return super(jax_discrete, cls).__new__(cls)
+
+    def __init__(self, *args, **kwargs):
+        self.is_logits = kwargs.pop("is_logits", False)
```

Comment:
Storing this for all distributions, i.e. those that don't even take in a probability param seems like a hack, but I guess there isn't a cleaner alternative here.